src/ansys/dpf/post/result_workflows/_sub_workflows.py

```diff
@@ -0,0 +1,232 @@
+from typing import Any, Union
+
+from ansys.dpf.core import Workflow
+from ansys.dpf.gate.common import locations
+
+from ansys.dpf.post.misc import _connect_any
+from ansys.dpf.post.result_workflows._utils import _CreateOperatorCallable
+from ansys.dpf.post.selection import _WfNames
+
+
+def _create_averaging_workflow(
+    has_skin: bool,
+    location: str,
+    force_elemental_nodal: bool,
+    create_operator_callable: _CreateOperatorCallable,
+    server,
+):
+    average_wf = Workflow(server=server)
+
+    input_data_fwd = create_operator_callable(name="forward_fc")
+    averaged_data_fwd = create_operator_callable(name="forward_fc")
+
+    mesh_averaging_input_fwd = create_operator_callable(name="forward_fc")
+    average_wf.add_operators(
+        [input_data_fwd, averaged_data_fwd, mesh_averaging_input_fwd]
+    )
+    mesh_averaging_input_fwd.connect(0, input_data_fwd, 0)
+
+    average_wf.set_input_name(_WfNames.input_data, input_data_fwd)
+    average_wf.set_output_name(_WfNames.output_data, averaged_data_fwd, 0)
+
+    skin_mesh_fwd = create_operator_callable(name="forward")
+
+    map_to_skin = True
```

Comment:
This has changed slightly. Before, the skin mapping never happened if force_elemental_nodal was false. This works for nodal data (where the skin mesh scoping only includes the skin nodes), but I believe it would fail for elemental data. In the new implementation, the mapping operator is always called. I had to add a server version check because the mapping operator did not support nodal data in older versions. (Fixed in this PR https://tfs.ansys.com:8443/tfs/ANSYS_Development/DPF/_git/DPF/pullrequest/450530)